### PR TITLE
PWN-9151 - MyCryptoFragment.kt created and logic added

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-9151

## Description of Work

PWN-9151 - MyCryptoFragment.kt created and logic added

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoFragment.kt [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            }
+            ActionButton.SWAP_BUTTON -> {
+                presenter.onSwapClicked()
+            }
+            else -> Timber.d("Unsupported Action! $clickedButton")
+        }
+    }
+
+    override fun showBalance(cellModel: TextViewCellModel?) {
+        binding.viewBalance.textViewAmount.bindOrGone(cellModel)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoFragment.kt:120]: balance can be empty if there is no usdc?

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoFragment.kt [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    override fun showReceive() {
+        replaceFragment(receiveFragmentFactory.receiveFragment())
+    }
+
+    override fun showSwap() {
+        replaceFragment(JupiterSwapFragment.create(source = SwapOpenedFrom.MAIN_SCREEN))
+    }
+
+    private fun onToggleClicked() {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoFragment.kt:148]: onVisibilityToggleClicked

on the first view i thought about feature toggle 😁

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoFragment.kt [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    private fun onHideClicked(token: Token.Active) {
+        homeAnalytics.logHiddenTokensClicked()
+        presenter.toggleTokenVisibility(token)
+    }
+
+    private fun onClaimTokenClicked(canBeClaimed: Boolean, token: Token.Eth) {
+        presenter.onClaimClicked(canBeClaimed, token)
+    }
+
+    override fun showTokenClaim(token: Token.Eth) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoFragment.kt:167]: navigateToTokenClaim

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            visibilityState = visibilityState,
+            isZerosHidden = areZerosHidden,
+        )
+        view?.showItems(mappedItems)
+    }
+
+    private fun getUserBalance(tokens: List<Token.Active>): BigDecimal {
+        if (tokens.none { it.totalInUsd != null }) return BigDecimal.ZERO
+
+        return tokens
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:108]: UserRepository already has this logic

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                    solTokens = newState.solTokens,
+                    ethTokens = newState.ethTokens
+                )
+            }
+        }
+    }
+
+    private fun showTokensAndBalance(solTokens: List<Token.Active>, ethTokens: List<Token.Eth>) {
+        val balance = getUserBalance(solTokens)
+        view?.showBalance(homeMapper.mapBalance(balance))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:92]: showBalance can get only non-null balance

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+            is UserTokensState.Empty -> {
+                view?.showEmptyState(isEmpty = true)
+                handleEmptyAccount()
+            }
+            is UserTokensState.Loaded -> {
+                view?.showEmptyState(isEmpty = false)
+                initializeActionButtons()
+                showTokensAndBalance(
+                    solTokens = newState.solTokens,
+                    ethTokens = newState.ethTokens
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:84]: Let's get rid of old functions like `initializeActionButtons()` or `showTokensAndBalance`

Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:84]: Can we try to make a single `CryptoState` which can modify the UI?
I want only one channel which edits the UI

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt (hunk 5 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:92]: yes that is true

File: app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/crypto/MyCryptoPresenter.kt:108]: yes but we will need to filter this tokens on USDC and other Wallet tokens in future

**Review by 1sakov — APPROVED** [review]

**Pull request merged**